### DISCUSSION TITLE
Modern build

### DISCUSSION THREAD

**Discussion by jods4** [exceljs/exceljs#1790] [Ideas & Proposals]
It'd be nice to include one more `dist` based on ES modules and without any polyfill nor old-JS transpilation.
This could be referenced by a `module` field in `package.json` so it's picked up by bundlers such as Webpack or Vite.

The main benefit here would be size, of course.
Exceljs is not small but there's a lot that could be easily removed for those who don't target IE.

- the current build includes helpers from every single file! One example: `_classCallCheck` declaration is copied 145 times in the output. The same goes for `_createClass`, `_defineProperties` and co.
- state machines generated in place of native `async/await` or `*/yield` functions are large.
- ES modules _could_ benefit from tree-shaking to trim unused features, although I suppose this may need some tweaks in the way API are built to really benefit from. In theory, it'd be nice if someone who only produces plain Excel files did not include any code related to CSV, parsing XLSX or using charts.

**Comment by baybal** [comment]
Exceljs doesn't work with Vite. The latest 4.3.0

**Reply by thomas-kahn** [reply]
I'm using Exceljs in Vite without any issues other than that it's a huge chunk of code. But that has nothing to do with Vite. 
The version of Vite I'm using is 4.0.4 and the version of Exceljs is 4.3.0.

**Reply by jods4** [reply]
Can confirm the same. 
Our project is built with Vite and uses Exceljs 4.3.0 without any issue (besides Exceljs being pretty big).

**Comment by cohlar** [comment]
Hi there, I'm wondering if anyone has found a lightweight (or tree-shaken) alternative to ExcelJS which allows to export **styled** Excel spreadsheets?
I've just added it to my Vite project, and it literally doubled my bundle size. I've tried [xlsx](https://www.npmjs.com/package/xlsx) which bundle size was a lot more decent, but styling spreadsheets is part of their paid tier 😶
